### PR TITLE
Added support for PHP instruction

### PR DESCRIPTION
**Pull request by ml85**
`
     PHA, PHP

        #  address R/W description
       --- ------- --- -----------------------------------------------
        1    PC     R  fetch opcode, increment PC
        2    PC     R  read next instruction byte (and throw it away)
        3  $0100,S  W  push register on stack, decrement S
`

### CODE REVIEW

File: core/src/mos6502.cpp [hunk 1]
```diff
@@ -74,6 +75,12 @@ void Mos6502::execute() {
             pipeline_.push(
                     [=]() { registers_->pc = mmu_->read_word(kBrkAddress); });
             return;
+        case PHP:
+            pipeline_.push([=]() { ++registers_->pc; });
+            pipeline_.push([=]() {
+                ram_.write_byte(--registers_->sp, registers_->p);
```
Comment (johnor) [inline core/src/mos6502.cpp]: Should the stack pointer be decremented after writing to the stack instead?

> push register on stack, decrement S

Comment (ml85) [reply core/src/mos6502.cpp]: I did some reading and from what I understand the 6502 uses an "empty stack" interpretation where the stack pointer shall always point to the location where the next value shall be stored. In that case we should decrement after pushing the value. What do you say about this @robinlinden ?

Comment (robinlinden) [reply core/src/mos6502.cpp]: Yeah, you're right. The 6502 uses an empty stack, but we should update all instructions to use an empty stack at the same time, so I'm fine with merging this while it uses a full stack like the rest of the library and fixing all the instructions in a different commit.

Comment (robinlinden) [reply core/src/mos6502.cpp]: I opened an issue about fixing this: #27

**Pull request merged**